### ISSUE TITLE
no-namespace: allow global augmentation

### ISSUE THREAD

**Issue by IlyaSemenov** [eslint/typescript-eslint-parser#477]
**What version of TypeScript are you using?**
2.8.3

**What version of `typescript-eslint-parser` are you using?**
15.0.0

**What code were you trying to parse?**

`index.d.ts`:

```ts
declare global {
	interface Process {
		client: boolean
		server: boolean
	}
}
```

or

```ts
declare namespace NodeJS {
	interface Process {
		client: boolean
		server: boolean
	}
}
```

with:

```js
rules: {
	'typescript/no-namespace': 'error',
},
```

**What did you expect to happen?**
No errors.

**What happened?**
`error  ES2015 module syntax is preferred over custom TypeScript modules and namespaces  typescript/no-namespace`

There was a similar issue in tslint - https://github.com/palantir/tslint/issues/3157 - and it was fixed.

**Comment by JamesHenry** [comment]
Thanks @IlyaSemenov, issues related to custom rules in the plugin should be reported there: https://github.com/nzakas/eslint-plugin-typescript

The rule will need to be updated.
